openedx/stanford/lms/lib/courseware_search/lms_filter_generator.py

```diff
@@ -0,0 +1,30 @@
+"""
+Custom override of SearchFilterGenerator without microsite org support
+using course tiles for discovery search
+"""
+from search.filter_generator import SearchFilterGenerator
+
+from branding_stanford.models import TileConfiguration
+from lms.lib.courseware_search.lms_filter_generator import LmsSearchFilterGenerator
+
+
+class TileSearchFilterGenerator(LmsSearchFilterGenerator):
+    """ SearchFilterGenerator for LMS Search """
+
+    def field_dictionary(self, **kwargs):
+        """ Return field filter dictionary for search """
+        # Adds course filter if provided, for in course search
+        field_dictionary = SearchFilterGenerator.field_dictionary(self, **kwargs)
+        if kwargs.get('user'):
+            if not kwargs.get('course_id'):
```

Comment:
What happens if there _is_ a `user` _and_ there _is_ a `course_id`?

What is the fall-through logic?

Reply:
The base class handles if there is a `course_id` specified, which means its an in-course search: https://github.com/edx/edx-search/blob/master/search/filter_generator.py#L27